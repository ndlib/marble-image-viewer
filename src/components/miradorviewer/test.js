import React from "react"
import renderer from "react-test-renderer"

import MiradorViewer from "./index.js"

describe("MiradorViewer", () => {
  it("renders usage when there is no manifest.", () => {
    const tree = renderer
      .create(<MiradorViewer id="test" location={{seach: ""}} themeColor="#red" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders a manifest when there is a manifest.", () => {
    const tree = renderer
      .create(<MiradorViewer id="test" location={{seach: ""}} themeColor="#red" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
