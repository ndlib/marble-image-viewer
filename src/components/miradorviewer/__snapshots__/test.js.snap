// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiradorViewer renders a manifest when there is a manifest. 1`] = `
<div
  className="sizeWrapper"
  style={Object {}}
>
  <div>
    <h1>
      Please add a query parameters of manifest="path/to/manifest" to display
    </h1>
    <p>
      Query String Options
      <dl>
        <dt>
          manifest
        </dt>
        <dd>
          The manifest url
        </dd>
        <dt>
          sidebar
        </dt>
        <dd>
          Sidebar open by default (true or false)
        </dd>
        <dt>
          thumbnails
        </dt>
        <dd>
          Thumbnail Navigation Position true === far-bottom false === off
        </dd>
        <dt>
          fullscreen
        </dt>
        <dd>
          Show full screen or not (true or false)
        </dd>
        <dt>
          canvasIndex
        </dt>
        <dd>
          Canvas index to start at
        </dd>
        <dt>
          view
        </dt>
        <dd>
          viewerView if not default
        </dd>
      </dl>
    </p>
  </div>
</div>
`;

exports[`MiradorViewer renders usage when there is no manifest. 1`] = `
<div
  className="sizeWrapper"
  style={Object {}}
>
  <div>
    <h1>
      Please add a query parameters of manifest="path/to/manifest" to display
    </h1>
    <p>
      Query String Options
      <dl>
        <dt>
          manifest
        </dt>
        <dd>
          The manifest url
        </dd>
        <dt>
          sidebar
        </dt>
        <dd>
          Sidebar open by default (true or false)
        </dd>
        <dt>
          thumbnails
        </dt>
        <dd>
          Thumbnail Navigation Position true === far-bottom false === off
        </dd>
        <dt>
          fullscreen
        </dt>
        <dd>
          Show full screen or not (true or false)
        </dd>
        <dt>
          canvasIndex
        </dt>
        <dd>
          Canvas index to start at
        </dd>
        <dt>
          view
        </dt>
        <dd>
          viewerView if not default
        </dd>
      </dl>
    </p>
  </div>
</div>
`;
